src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -184,10 +184,22 @@ double FFN<OutputLayerType, InitializationRuleType>::Evaluate(
   currentTarget = responses.unsafe_col(i);
 
   Forward(std::move(currentInput));
-  double res = outputLayer.Forward(std::move(boost::apply_visitor(
+  currentCost = outputLayer.Forward(std::move(boost::apply_visitor(
       outputParameterVisitor, network.back())), std::move(currentTarget));
 
-  return res;
+  return currentCost;
+}
+
+template<typename OutputLayerType, typename InitializationRuleType>
+double FFN<OutputLayerType, InitializationRuleType>::Evaluate(
```

Comment:
It depends on what you expect if I optimize something using a single iteration what I get is the performance before I updated the parameters right? That's why the gradient checks are failing.